tools/python/boutiques/tests/test_example1.py

```diff
@@ -71,6 +77,34 @@ def test_example1_exec_docker(self):
         assert(ret.output_files[0].file_name == "log-4-coin;plop.txt" or
                ret.output_files[1].file_name == "log-4-coin;plop.txt")
 
+    @pytest.mark.skipif(subprocess.Popen("type docker", shell=True).wait(),
+                        reason="Docker not installed")
+    def test_example1_exec_docker_stream_output(self):
+        example1_dir = os.path.join(self.get_examples_dir(), "example1")
+        self.clean_up()
+        ret = bosh.execute("launch",
+                           os.path.join(example1_dir,
+                                        "example1_docker.json"),
+                           "-s",
+                           os.path.join(example1_dir,
+                                        "invocation.json"))
+
+        # Make sure stdout and stderr are printed on the fly for
+        # streaming mode
+        out, err = self.capfd.readouterr()
```

Comment:
Sorry, I don't understand what this does :) Could you add a line to explain where readouterr() comes from?

Reply:
`capfd` is a Pytest fixture that captures the stdout and stderr during test execution and returns them as a tuple in `readouterr()`. I'll add another comment in the file explaining this.

https://docs.pytest.org/en/latest/capture.html